File: Challenges/Coding World Cup/2023/Day 10 - IND vs PAK - Rest Days.py
from typing import List


def ninjaWantsHoliday(n: int, k: int, a: List[int]) -> str:
    # Write your code here.
    a.sort()
    count = 1
    start = a[0]

    for i in range(1, n):
        if start + 1 == a[i]:
            count += 1
            start = a[i]
        else:
            start = a[i]
            count = 1

        if count == k:
            return "YES"

    return "NO"

File: Challenges/Coding World Cup/2023/Day 10 - IND vs PAK - Rest Days.java
import java.util.Arrays;

public class Solution {
    static String ninjaWantsHoliday(int n, int k, int []a) {
        Arrays.sort(a);
        int count = 1;
        int start = a[0];

        for (int i = 1; i < n; i++) {
            if (start + 1 == a[i]) {
                count += 1;
                start = a[i];
            } else {
                start = a[i];
                count = 1;
            }

            if (count == k) {
                return "YES";
            }
        }

        return "NO";
    }
}